From linux-kernel-owner@vger.kernel.org Mon Oct 20 16:01:15 2003
Return-Path: <workshop@cpt.saharapc.co.za>
X-Original-To: azarah@tamperd.net
Delivered-To: azarah@tamperd.net
Received: from localhost (localhost [127.0.0.1]) by mail1.tamperd.net
	(Postfix) with ESMTP id B3F3C26C0FA for <azarah@tamperd.net>; Mon, 20 Oct
	2003 16:01:15 +0200 (CEST)
Received: from emu.gentoo.org (emu.gentoo.oregonstate.edu [128.193.0.39])
	by mail1.tamperd.net (Postfix) with ESMTP id 9ED4726C0E6 for
	<azarah@tamperd.net>; Mon, 20 Oct 2003 16:01:14 +0200 (CEST)
Received: from mail.gentoo.org ([204.126.2.42] helo=gentoo.org) by
	emu.gentoo.org with esmtp (Exim 4.22) id 1ABaav-00048k-T9 for
	azarah@dev.gentoo.org; Mon, 20 Oct 2003 14:01:14 +0000
Received: (qmail 9395 invoked by alias); 20 Oct 2003 14:01:13 -0000
Delivered-To: azarah@gentoo.org
Received: (qmail 2656 invoked from network); 20 Oct 2003 14:01:07 -0000
Received: from unknown (HELO angel.sistech.co.za) (196.41.3.119) by
	mail.gentoo.org with SMTP; 20 Oct 2003 14:01:07 -0000
Received: from andromeda.cpt.sahara.co.za (mail.cpt.sahara.co.za
	[196.41.29.142]) by angel.sistech.co.za (8.11.6/SVW2.4) with ESMTP id
	h9KE0uv12906 for <azarah@gentoo.org>; Mon, 20 Oct 2003 16:00:57 +0200
Received: from tom.saharacpt.lan ([10.0.4.50]) by
	andromeda.cpt.sahara.co.za (8.11.6/8.11.2/SuSE Linux 8.11.1-0.5) with ESMTP
	id h9KA91518467 for <azarah@gentoo.org>; Mon, 20 Oct 2003 12:09:07 +0200
Resent-Date: Mon, 20 Oct 2003 12:09:07 +0200
Resent-Message-Id: <200310201009.h9KA91518467@andromeda.cpt.sahara.co.za>
Received: from vger.kernel.org (vger.kernel.org [67.72.78.212]) by
	andromeda.cpt.sahara.co.za (8.11.6/8.11.2/SuSE Linux 8.11.1-0.5) with ESMTP
	id h9HIX5522452 for <workshop@cpt.saharapc.co.za>; Fri, 17 Oct 2003
	20:33:06 +0200
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand id
	S263494AbTJQScQ (ORCPT <rfc822;workshop@cpt.saharapc.co.za>); Fri, 17 Oct
	2003 14:32:16 -0400
Received: (majordomo@vger.kernel.org) by vger.kernel.org id S263583AbTJQScQ
	(ORCPT <rfc822;linux-kernel-outgoing>); Fri, 17 Oct 2003 14:32:16 -0400
Received: from mail.kroah.org ([65.200.24.183]:27032 "EHLO
	perch.kroah.org") by vger.kernel.org with ESMTP id S263494AbTJQScO (ORCPT
	<rfc822;linux-kernel@vger.kernel.org>); Fri, 17 Oct 2003 14:32:14 -0400
Received: from DYN318258BLD.beaverton.ibm.com (bi01p1.co.us.ibm.com
	[32.97.110.142]) (authenticated) by perch.kroah.org (8.11.6/8.11.6) with
	ESMTP id h9HIQUh14125; Fri, 17 Oct 2003 11:26:30 -0700
Received: from greg by echidna.kroah.org with local (masqmail 0.2.19) id
	1AAZKM-2nK-00; Fri, 17 Oct 2003 11:27:54 -0700
From: Greg KH <greg@kroah.com>
To: clemens@dwf.com
Cc: linux-hotplug-devel@lists.sourceforge.net, linux-kernel@vger.kernel.org, reg@orion.dwf.com
Message-ID: <20031017182754.GA10714@kroah.com>
References: <20031017055652.GA7712@kroah.com>
	 <200310171757.h9HHvGiY006997@orion.dwf.com>
	 <20031017181923.GA10649@kroah.com>
Mime-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline
In-Reply-To: <20031017181923.GA10649@kroah.com>
User-Agent: Mutt/1.4.1i
Sender: linux-kernel-owner@vger.kernel.org
Precedence: bulk
X-Mailing-List: 	linux-kernel@vger.kernel.org
X-SAHARACPT-MailScanner: Found to be clean, Found to be clean
X-UIDL: kRp"!<XG!!T5$"!~DB!!
Subject: Re: [ANNOUNCE] udev 003 release
Resent-From: Sahara Workshop <workshop@cpt.saharapc.co.za>
Resent-To: Martin Schlemmer <azarah@gentoo.org>
X-Mailer: Ximian Evolution 1.4.4 
Date: Mon, 20 Oct 2003 12:07:06 +0200
X-SAHARACPT-MailScanner-Information: Please contact the ISP for more
	information
X-ANGEL-MailScanner-Information: Please contact the ISP for more information
X-ANGEL-MailScanner: Found to be clean
Resent-Date: Mon, 20 Oct 2003 14:01:14 +0000
X-Virus-Scanned: by AMaViS 0.3.12
X-Evolution-Source: pop://azarah%40tamperd.net@lajka.tamperd.net
Content-Transfer-Encoding: 8bit

On Fri, Oct 17, 2003 at 11:19:23AM -0700, Greg KH wrote:
> 
> Ah, yeah, udev seg faults right now for partitions.  Let me try to track
> down the bug, give me a bit of time...

Here's a patch that fixes the partition logic for me.  Sorry about this, I
need to make sure to test partitions more next time.

thanks again for testing this.

greg k-h

# fix segfaults when dealing with partitions.

diff -Nru a/udev-add.c b/udev-add.c
--- a/udev-add.c	Fri Oct 17 11:26:37 2003
+++ b/udev-add.c	Fri Oct 17 11:26:37 2003
@@ -169,13 +169,13 @@
 	}
 	memset(&dbdev, 0, sizeof(dbdev));
 	strncpy(dbdev.name, attr.name, NAME_SIZE);
-	strncpy(dbdev.sysfs_path, class_dev->sysdevice->directory->path,
-		PATH_SIZE);
+	if (class_dev->sysdevice) {
+		strncpy(dbdev.sysfs_path, class_dev->sysdevice->directory->path, PATH_SIZE);
+		strncpy(dbdev.bus_id, class_dev->sysdevice->bus_id, ID_SIZE);
+	}
 	strncpy(dbdev.class_dev_name, class_dev->name, NAME_SIZE);
-	if ((sysfs_get_name_from_path(subsystem, dbdev.class_name, NAME_SIZE))
-	    != 0)
+	if ((sysfs_get_name_from_path(subsystem, dbdev.class_name, NAME_SIZE)) != 0)
 		strcpy(dbdev.class_name, "unkown");
-	strncpy(dbdev.bus_id, class_dev->sysdevice->bus_id, ID_SIZE);
 	strcpy(dbdev.bus_name, "unknown");
 	if (class_dev->driver != NULL)
 		strncpy(dbdev.driver, class_dev->driver->name, NAME_SIZE);
-
To unsubscribe from this list: send the line "unsubscribe linux-kernel" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html
Please read the FAQ at  http://www.tux.org/lkml/


